### PR TITLE
fix: qwen3_8b_hellaswag_pp_peft recipe

### PR DESCRIPTION
**Pull request by ZhiyuLi-Nvidia**
# What does this PR do ?

As titled, remove explicitly set cache_dir and add pad sequence length.

# Changelog

- Add specific line by line info of high level changes in this PR.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: examples/llm_finetune/qwen/qwen3_8b_hellaswag_pp_peft.yaml [hunk 1]
```diff
@@ -82,7 +82,9 @@ packed_sequence:
 
 dataloader:
   _target_: torchdata.stateful_dataloader.StatefulDataLoader
-  collate_fn: nemo_automodel.components.datasets.utils.default_collater
+  collate_fn:
+    _target_: nemo_automodel.components.datasets.utils.default_collater
+    pad_seq_len_divisible: 320
```
Comment (akoumpa) [inline examples/llm_finetune/qwen/qwen3_8b_hellaswag_pp_peft.yaml]: Hi @ZhiyuLi-Nvidia. for this one can you share a little more context why it is needed? I thought for fp8 we needed the padding, but I'm not sure what I'm missing here.

Comment (ZhiyuLi-Nvidia) [reply examples/llm_finetune/qwen/qwen3_8b_hellaswag_pp_peft.yaml]: The root cause is torch pp doesn't support variant seq len. This is to do the padding to avoid the above issue.

Could you take a look at the logs: https://wandb.ai/nvidia/automodel-dev-zhiyul/runs/0t1oh7bz/logs

```
torchrun --nproc-per-node=8 nemo_automodel/recipes/llm/train_ft.py --config examples/llm_finetune/qwen/qwen3_8b_hellaswag_pp_peft.yaml 
```

Without the change, The pipeline compiled based on the constant cached the shape [2, 168, 4096] from training step.
When validation started, the first validation batch happened to be shorter (seq_len=72), resulting in a shape of [2, 72, 4096]. The pipeline validation logic detected that 72 != 168 and raised the PipeliningShapeError.
